tests/unit/test_space.py

```diff
@@ -212,3 +212,12 @@ def test_box_discretize_returns_search_space_with_correct_number_of_points(
 
     with pytest.raises(ValueError):
         dss.sample(num_samples + 1)
+
+
+def test_box_combined_with_itself_returns_new_box_with_bounds_twice_as_large(
+) -> None:
+    box = Box(tf.zeros((3,)), tf.ones((3,)))
+    new_box = box * box
+
+    assert len(new_box.lower) == 6
+    assert len(new_box.upper) == 6
```

Comment:
how about a test that checks the contents of the lower and upper bounds for two `Box`es
```
box1 = Box(tf,constant([0., 1.])), tf.constant([2., 3.])
box2 = Box(tf,constant([4., 5.])), tf.constant([6., 7.])

res = box1 * box2
npt.assert_allclose(res.lower, [0, 1, 4, 5])
npt.assert_allclose(res.upper, [2, 3, 6, 7])
```

Reply:
and also one with different dtypes, that it errors, with
```
box1 = Box(tf.constant([0., 1.])), tf.constant([2., 3.])
box2 = Box(tf.constant([4., 5.], tf.float64)), tf.constant([6., 7.], tf.float64)

with pytest.raises(TypeError):
    _ = box1 * box2
```